src/core/collections.js

```diff
@@ -59,11 +59,27 @@ export const initCollection = screen => key => {
 		gmi.setGameData("genie", fp.setWith(Object, storagePath, mergeToList(getStored(), [config]), getGenieStore()));
 	};
 
+	const setUnique = ({ id, key, value }) => {
+		catalogue.forEach(item => {
+			const isNewlyUnique = item.id === id && item[key] !== value;
+			const shouldBeReset = item.id !== id && item.hasOwnProperty(key) && item[key] === value;
+
+			if (isNewlyUnique) {
+				set({ id, [key]: value });
+				item[key] = value;
+			} else if (shouldBeReset) {
+				set({ id: item.id, [key]: null });
+				item[key] = null;
+			}
```

Comment:
would this work if it was simplified to just update the value if

```
item.id === id && item[key]
```

and set to null if not?

Reply:
not suggesting you do this, but the following seems to work, with the caveat of adding 'null' to all non-selected items:

```suggestion
			const isNewlyUnique = item.id === id;

			if (isNewlyUnique) {
				set({ id, [key]: value });
				item[key] = value;
			} else {
				set({ id: item.id, [key]: null });
				item[key] = null;
			}
```

I guess that was what you were trying to avoid?

Reply:
it's a shame we can't unset the value - it would be much cleaner in the saveData object if we could set undefined.

Reply:
We considered this with Nick but then decided against it so we don't append an unnecessary `[key]: null` to each element that doesn't need it.